AnkiHubSoftware/ankihub_addon
# fix: Don't use progress dialog as parent of dialogs

This PR fixes an issue where a progress dialog is used as a parent of another dialog. This is problematic, because the progress dialog is typically closed shortly after the other dialog is shown, which leads to the other dialog being closed as well.  This problem occurs for example when trying to sync without a membership and a message should be shown to the user explaining why they can't sync.

## Related issues


## Proposed changes
- Add a `active_window_or_mw` utility function:
  https://github.com/ankipalace/ankihub_addon/blob/526e524b7528f041f31fbafb36a2b1b7d5996a72/ankihub/gui/utils.py#L571-L585
- Use this function everywhere where we are currently using `aqt.mw.app.activeWindow()` for getting a parent window for a dialog.


## How to reproduce
- Go to django admin and set your membership to inactive
- Sync with AnkiHub from Anki
- Observe that the dialog (shown below) stays open
- Repeat on the main branch and observe that the dialog is quickly closed


## Screenshots and videos
https://github.com/ankipalace/ankihub_addon/assets/31575114/6c24f46f-d9a3-4f29-a1fd-746699b7c0d1

<img src="https://github.com/ankipalace/ankihub_addon/assets/31575114/5c3c2d9e-c210-4a5a-b0c1-af02a9270fdb" width="500" />
